### PR TITLE
fix rainbow breaks compound emojis

### PR DESCRIPTION
**Pull request by EECvision**
### Bug Fixes:

Fixes https://github.com/vector-im/element-web/issues/21371

After:
error

Signed-off-by: Emmanuel Ezeka [eec.studies@gmail.com](mailto:eec.studies@gmail.com)

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * fix rainbow breaks compound emojis ([\#8245](https://github.com/matrix-org/matrix-react-sdk/pull/8245)). Fixes vector-im/element-web#21371. Contributed by @EECvision.

Preview: https://pr8245--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by turt2live — CHANGES_REQUESTED** [review]
Our font families can be changed by users, so this wouldn't work for them. A more detailed look into the problem would be appreciated, as we have code which shouldn't be trying to split the emoji in two like that.

A unit test to verify the behaviour and prevent the bug from resurfacing again would also be appreciated, as this will be the second time we've fixed this bug.

**Comment by EECvision** [comment]
> Our font families can be changed by users, so this wouldn't work for them. A more detailed look into the problem would be appreciated, as we have code that shouldn't be trying to split the emoji in two like that.
> 
> A unit test to verify the behaviour and prevent the bug from resurfacing again would also be appreciated, as this will be the second time we've fixed this bug.

Hi, I am still looking for a permanent solution to the issue. Do you think that applying rainbow to only text but not on emoji will be ok?

**Comment by turt2live** [comment]
Some emoji support rainbows, so we should keep them as rainbows. However, we shouldn't be splitting emoji apart to do that.

**Comment by EECvision** [comment]
I have used lodash `split` method which prevents splitting emojis in two. I have also tested the behaviour. However, I know that adding `font-family: Segoe UI Symbol;` is not valid since users need to change `font-family` dynamically. I will appreciate any idea on how I can apply `rainbow command` to all emojis without manually setting a `font-family`. Thank you.

File: res/css/views/elements/_RichText.scss [hunk 1]
```diff
@@ -80,6 +80,7 @@ a.mx_Pill {
     // and scale with the size of the surrounding text
     font-size: calc(18 / 14 * 1em);
     vertical-align: bottom;
+    font-family: Segoe UI Symbol;
```
Comment (robintown) [inline res/css/views/elements/_RichText.scss:83]: I'm not really concerned with having the emoji themselves be rainbow, and we really shouldn't switch from Twemoji to a font that is only available on Windows systems, so please remove this change.

File: test/utils/colour-test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2021 Šimon Brandner <simon.bra.ag@gmail.com>
```
Comment (robintown) [inline test/utils/colour-test.ts:2]: Thanks for writing this test! This should be your own copyright here

File: res/css/views/elements/_RichText.scss (hunk 1 above)
Comment (EECvision) [reply res/css/views/elements/_RichText.scss]: Okay, I have removed the change.

**Review by robintown — APPROVED** [review]
Looks good now, thanks :+1:

**Pull request merged**